account.go

```diff
@@ -34,6 +34,121 @@ func (d Distribution) String() string {
 	return fmt.Sprintf("Distribution{%s, %sregen}", d.Time.Format(time.RFC3339), d.Regen.String())
 }
 
+var Dec128Context = apd.Context{
+	Precision:   34,
+	MaxExponent: apd.MaxExponent,
+	MinExponent: apd.MinExponent,
+	Traps:       apd.DefaultTraps,
+}
+
+func RecordToAccount(rec Record, genesisTime time.Time) (Account, error) {
+	amount := rec.TotalAmount
+	distTime := rec.StartTime
+	if distTime.IsZero() {
+		distTime = genesisTime
+	}
+	numDist := rec.NumMonthlyDistributions
+	if numDist < 1 {
+		numDist = 1
+	}
+
+	if numDist == 1 && !distTime.After(genesisTime) {
+		return Account{
+			Address:    rec.Address,
+			TotalRegen: amount,
+			Distributions: []Distribution{
+				{
+					Time:  genesisTime,
+					Regen: amount,
+				},
+			},
+		}, nil
+	}
+
+	distAmount, dust, err := distAmountAndDust(amount, numDist)
+	if err != nil {
+		return Account{}, err
+	}
+
+	var distributions []Distribution
+	var genesisAmount apd.Decimal
+
+	// collapse all pre-genesis distributions into a genesis distribution
+	for ; numDist > 0 && !distTime.After(genesisTime); numDist-- {
+		_, err = apd.BaseContext.Add(&genesisAmount, &genesisAmount, &distAmount)
+		if err != nil {
+			return Account{}, err
+		}
+		distTime = distTime.Add(OneMonth)
+	}
+
+	// if there is a genesis distribution add it
+	if !genesisAmount.IsZero() {
+		distributions = append(distributions, Distribution{
+			Time:  genesisTime,
+			Regen: genesisAmount,
+		})
+	}
+
+	// add post genesis distributions
+	for ; numDist > 0; numDist-- {
+		distributions = append(distributions, Distribution{
+			Time:  distTime,
+			Regen: distAmount,
+		})
+		distTime = distTime.Add(OneMonth)
+	}
+
+	// add dust to first distribution
+	_, err = apd.BaseContext.Add(&distributions[0].Regen, &distributions[0].Regen, &dust)
+	if err != nil {
+		return Account{}, err
+	}
+
+	return Account{
+		Address:       rec.Address,
+		Distributions: distributions,
+		TotalRegen:    amount,
+	}, nil
+}
+
+func distAmountAndDust(amount apd.Decimal, numDist int) (distAmount apd.Decimal, dust apd.Decimal, err error) {
+	if numDist <= 1 {
+		return amount, dust, nil
+	}
+
+	var numDistDec apd.Decimal
+	numDistDec = *numDistDec.SetInt64(int64(numDist))
+
+	_, err = Dec128Context.Mul(&amount, &amount, tenE6)
```

Comment:
```suggestion
	// convert amount from regen to uregen, so we can perform integral arithmetic on uregen
	_, err = Dec128Context.Mul(&amount, &amount, tenE6)
```

Looks like this is coverting amount from regen => uregen. Can we name the variables more explicitly to make that clear? amtRegen and amtUregen or similar? Alternatively, atleast we sould provide a comment here.